cloudinit/net/dhcp.py

```diff
@@ -281,24 +303,30 @@ def parse_static_routes(rfc3442):
     """ parse rfc3442 format and return a list containing tuple of strings.
 
     The tuple is composed of the network_address (including net length) and
-    gateway for a parsed static route.
+    gateway for a parsed static route.  It can parse two formats of rfc3442,
+    one from dhcpcd and one from dhclient (isc).
 
-    @param rfc3442: string in rfc3442 format
+    @param rfc3442: string in rfc3442 format (isc or dhcpd)
     @returns: list of tuple(str, str) for all valid parsed routes until the
               first parsing error.
 
     E.g.
-    sr = parse_state_routes("32,169,254,169,254,130,56,248,255,0,130,56,240,1")
-    sr = [
+    sr=parse_static_routes("32,169,254,169,254,130,56,248,255,0,130,56,240,1")
+    sr=[
         ("169.254.169.254/32", "130.56.248.255"), ("0.0.0.0/0", "130.56.240.1")
     ]
 
+    sr2 = parse_static_routes("24.191.168.128 192.168.128.1,0 192.168.128.1")
+    sr2 = [
+        ("191.168.128.0/24", "192.168.128.1"), ("0.0.0.0/0", "192.168.128.1")
+    ]
+
     Python version of isc-dhclient's hooks:
        /etc/dhcp/dhclient-exit-hooks.d/rfc3442-classless-routes
     """
     # raw strings from dhcp lease may end in semi-colon
     rfc3442 = rfc3442.rstrip(";")
-    tokens = rfc3442.split(',')
+    tokens = [tok for tok in re.split(r"[, .]",rfc3442) if tok]
```

Comment:
```suggestion
    tokens = [tok for tok in re.split(r"[, .]", rfc3442) if tok]
```
pycodestyle failed, need a space after the first argument to re.split()

Reply:
sorry, must have introduced this when working in the train...  just pushed up the fix.